### PR TITLE
[POC] Search box

### PR DESCRIPTION
**Pull request by MadameSheema**
Solves: https://github.com/franksparks/BookJourney/issues/6

### Update

- Dynamic pagination has been implemented taking the total number of items of the first response.
- We have extracted all the search bar logic into its own component `SearchBar`
- As output of this PR a new component for search has been created but is not used yet, it will be used on follow-up PRs.

### Context
In this PR we are investigating how to implement a simple search box that uses google Books API.

To do so we are using the following built-in components:
- [Shadcn](https://ui.shadcn.com/) for the input and the table.
- [Mui](https://mui.com/) for the pagination.

### Known/found limitations

* During the development of this POC I found out that the result was not always consistent, sometimes everything was in upper-case, others not, so a function that formats the output to be consistent with the results we display was created.

* Currently the pagination is hardcoded to 10. This is because I found out the API response was not consistent for the `totalItems` value.

* I know that I'm not good at naming so any improvement is welcomed.

### How to test it

- Checkout the PR
- Run
```cli
bun install
``` 
- Run
```cli
bun run dev
``` 
- Navigate in your browser to `http://localhost:3000`

### CODE REVIEW

File: components/SearchPagination.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Pagination from "@mui/material/Pagination";
+import { useState } from "react";
```
Comment (franksparks) [inline components/SearchPagination.tsx:2]: Let's remove this unused import

File: components/SearchResults.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export default function SearchResult({ results }: SearchResultProps) {
+    return (
+        <Table className="mt-5">
+            <TableBody>
+                {results.map((result, index) => (
+
+                    <TableRow key={index}>
+                        <>{result.smallThumbnail && <TableCell><img src={result.smallThumbnail} /></TableCell>}
```
Comment (franksparks) [inline components/SearchResults.tsx:16]: Note for the future: 
Change "img" HTML tags by shadcn "Image" component 

Should we add a TODO comment here?

Comment (MadameSheema) [reply components/SearchResults.tsx:16]: Which is the exact component? I'm not able to find the image one on the shadcn page.

**Review by franksparks — APPROVED** [review]
Application can be launched and works as expected.

(Note: I needed to delete _bun.lockb_ file and recreate it)

**Pull request merged**